### ISSUE TITLE
When stdin is a terminal, for_byte_record_with_terminator() does not exit immediately on Control-D

### ISSUE THREAD

**Issue by zhitkoff** [BurntSushi/bstr#180]
```rust
if stdin().is_terminal() {
        let mut buf_in = BufReader::new(stdin());
        let separator = b'\n';

        println!("Start terminal input for bstr for_byte_record_with_terminator\nTry ^D - it will not exit right away on the first ^D");
        let _result_bstr = buf_in.for_byte_record_with_terminator(separator, |_line| {
            // do something with each line
            Ok(true)
        })?;
        println!("Done");

        println!("Start terminal input for std io, like split, read_until, etc\nTry ^D");
        let _result_std = buf_in.split(separator).for_each(|_line| {
            // do something with each line
        });
        println!("Done");
    }
```

Comparing to similar functions from std::io, which do exit on the first Control-D right away in the terminal.
This limits usefulness of bstr crate for developing command line utilities that need to process all stdin variants (piped, redirected from file, terminal)

**Comment by BurntSushi** [comment]
Please provide a full program and an example invocation on the CLI that reproduces the problem. Please include actual output and expected output.

**Comment by zhitkoff** [comment]
Full program for bstr use case:
```rust
use bstr::io::BufReadExt;
use std::io::{self, stdin, BufReader, IsTerminal};

fn main() -> io::Result<()> {
    if stdin().is_terminal() {
        let mut buf_in = BufReader::new(stdin());
        let separator = b'\n';

        println!("Start terminal input for bstr for_byte_record_with_terminator\nTry ^D - it will not exit right away on the first ^D");
        let _result_bstr = buf_in.for_byte_record_with_terminator(separator, |_line| {
            // do something with line
            Ok(true)
        })?;
        println!("Done");
    }
    Ok(())
}
```

Run with `cargo run` and it will wait for user input in the terminal. Try `^D` (Control-D) and notice that it does not exist, still accepting input in the terminal. Try `^D` twice - it should exit to command prompt

**Comment by zhitkoff** [comment]
Full program to illustrate that using `std::io` functions in the same scenario does exist on the first `^D`:
```rust
use std::io::{self, stdin, BufRead, BufReader, IsTerminal};

fn main() -> io::Result<()> {
    if stdin().is_terminal() {
        let buf_in = BufReader::new(stdin());
        let separator = b'\n';

        println!("Start terminal input for std io, like split, read_until, etc\nTry ^D");
        let _result_std = buf_in.split(separator).for_each(|_line| {
            // do something with line
        });
        println!("Done");
    }
    Ok(())
}
```

Run with `cargo run` and it will wait for user input in the terminal. Try `^D` (Control-D) once and notice that it does exit into command prompt right away.

**Comment by BurntSushi** [comment]
Nice find. This is fixed in `bstr 1.9.1`.
